folding-schemes/src/folding/nova/decider_eth_circuit.rs

```diff
@@ -416,10 +461,78 @@ where
             >::check(cf_r1cs, cf_W_i.E, cf_U_i.u.clone(), cf_z_U)?;
         }
 
+        // 6. check KZG challenges
+        let (incircuit_c_W, incircuit_c_E) = KZGChallenges::<C1>::get_challenges_gadget(
+            cs.clone(),
+            &self.poseidon_config,
+            U_i1.clone(),
+        )?;
+        incircuit_c_W.enforce_equal(&kzg_c_W)?;
+        incircuit_c_E.enforce_equal(&kzg_c_E)?;
+
         Ok(())
     }
 }
 
+/// Gadget that computes the KZG challenges, also offers the rust native implementation compatible
+/// with the gadget.
+pub struct KZGChallenges<C: CurveGroup> {
```

Comment:
Right, maybe it can be added `Gadget` at the end but without the `Compute` at the beginning? ie. `KZGChallengesGadget`, how would you see that?
In this way it would be following the naming used for other similar gadgets such as `ChallengeGadget`, `CycleFoldChallengeGadget`, etc
(updated [here](https://github.com/privacy-scaling-explorations/folding-schemes/compare/167bc813a978d862029ae37c7a7d496a84a2f66f..63bcdae7d1675599b8d0b86443d677f1f4f09d67) )